pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -222,12 +222,17 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 		}
 	}
 
-	// Find the kserve-container (this is the model inference server) and transformer container
+	// Find the kserve-container (this is the model inference server) and transformer container and the worker-container
 	userContainer := getContainerWithName(pod, constants.InferenceServiceContainerName)
 	transformerContainer := getContainerWithName(pod, constants.TransformerContainerName)
+	workerContainer := getContainerWithName(pod, constants.WorkerContainerName)
 
 	if userContainer == nil {
-		return fmt.Errorf("Invalid configuration: cannot find container: %s", constants.InferenceServiceContainerName)
+		if workerContainer == nil {
+			return fmt.Errorf("Invalid configuration: cannot find container: %s", constants.InferenceServiceContainerName)
+		} else {
+			userContainer = workerContainer
```

Comment:
maybe we need a different name here, userContainer is not the same thing, see line 226.
Better to keep it segregated from each other.

Reply:
This is the correct one. The logic works as follows: Previously, it only checked for the `kserve-container`, but now it will also check for the presence of a `worker-container` if the `kserve-container` does not exist

Reply:
I am not referring to the logic, but the env used, the work-container is not the user-container.

Reply:
To me, it is ok but I am also ok with another name
so I choose `targetContainer` for it. please check it out.

Reply:
No, that's not quite correct. Our initial proposal is to support only the PVC protocol, and it's the user's responsibility to prepare the PVC. Therefore, the storage-initializer is required for both deployments because it needs to create a symbolic link from the folder containing the model to a specific directory.

Reply:
Let's not assume only head node has the model, user can have local storage and may not necessarily has the network disks